### DISCUSSION TITLE
Error with uWs.HttpResponse

### DISCUSSION THREAD

**Discussion by timfemey** [kartikk221/hyper-express#92] [Q&A] [answered]
```
 this.#raw_response.pause();
                               ^

Error: Invalid access of discarded (invalid, deleted) uWS.HttpResponse/SSLHttpResponse.
```

I have written scripts in place to handle user sending wrong field name and mimetype but if a user sends a wrong field name or mimetype i get the error below. Code below:

```
try {
  await req.multipart(
    {
      limits: { files: 1, fileSize: 2 * 1024 * 1024 },
      headers: { "content-type": req.headers["content-type"] },
    },
    async (field) => {
      if (field.name && field.name === "file") {
        if (
          field.mime_type == "image/png" ||
          field.mime_type == "image/jpeg" ||
          field.mime_type == "image/jpg" ||
          field.mime_type == "image/webp" ||
          field.mime_type == "image/svg" ||
          field.mime_type == "image/gif" ||
          field.mime_type == "image/jiff"
        ) {
          if (field.file) {
            //@ts-ignore
            stream2buffer(field.file.stream).then((fileBuf: Buffer) => {
              //Do stuff like resizing with sharp module and upload to File Server                   
            });
          } else {
            res.json({ message: "No File received", status: false });
          }
        } else {
          res.status(400).json({
            message: "Unsupported File Format sent",
            status: false,
          });
        }
      } else {
        res.status(401).json({ message: "Invalid Field", status: false });
      }
    }
  );
} catch (error) {
  console.log("Here");
  console.error(typeof error);
}
```

**Comment by kartikk221** [comment]
The the invalid discarded access error bug should be fixed in the recently released v6.2.6 of HyperExpress. See if that resolves your issue.

**Reply by timfemey** [reply]
It still return the same error after upgrade to V6.2.6

**Comment by BobbyWibowo** [comment, marked as answer]
I also got the error in 6.2.6
```
Uncaught Exception: Error: Invalid access of discarded (invalid, deleted) uWS.HttpResponse/SSLHttpResponse.
    at Request.pause (/home/bobby/src/safe.fiery.me/node_modules/hyper-express/src/components/http/Request.js:109:32)
    at Request.Readable.unpipe (node:internal/streams/readable:865:10)
    at unpipe (node:internal/streams/readable:796:9)
    at Multipart.onerror (node:internal/streams/readable:765:5)
    at Multipart.emit (node:events:539:35)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

Here's a simplification of my actual backend:
```js
await req.multipart(async field => {
    if (!field.file) return

    const readStream = field.file.stream
    const writeStream = fs.createWriteStream('/path/to/tmp')
    readStream.pipe(writeStream)
  })
```

The error throws when the request is still open, because the client is still uploading their multiform POST, but then said client refreshes the browser tab in which they were still uploading (i.e. they just force close the connection)

I can replicate it every time

**Reply by kartikk221** [reply]
Oh I see, yeah so this error is likely occuring due to uWebsockets.js having differing lifecycles for the `uWS.HttpRequest` and `uWS.HttpResponse` components. Essentially, the `uWS.HttpRequest` component lives for only first synchronous execution. The `uWS.HttpResponse` lives until the response is sent or the connection is closed. But the odd behavior is that the `uWS.HttpResponse.onData()` handler which is responsible for providing incoming body chunks continues to pipe data even after the response has been sent. The last scenario is where If you try and access `uWS.HttpResponse` after a response has been sent from the `onData` handler, the invalid access error will occur. I'll try and recreate the bug with the browser and have a fix rolled out for the next update.

**Reply by kartikk221** [reply]
The above problem should be fixed in the recently released v6.3.0 version of HyperExpress.

**Reply by timfemey** [reply]
Yes, it works now in v6.3.0  

> The above problem should be fixed in the recently released v6.3.0 version of HyperExpress.
